reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -4029,6 +4029,50 @@ public final Flux<T> concatWith(Publisher<? extends T> other) {
 		return concat(this, other);
 	}
 
+	/**
+	 * If Micrometer's {@code context-propagation} library is on the classpath, this
+	 * is a convenience shortcut to capture thread local values during the subscription phase
+	 * and put them in the {@link Context} that is visible upstream of this operator.
+	 * <p>
+	 * As a result this operator should generally be used as close as possible to the end of
+	 * the chain / subscription point.
+	 *
+	 * @return a new {@link Flux} where context-propagation API has been used to capture entries and
+	 * inject them into the {@link Context}
+	 * @see #contextCapture(Predicate)
+	 */
+	public final Flux<T> contextCapture() {
+		if (!ContextPropagation.isContextPropagationAvailable()) {
+			return this;
+		}
+		return onAssembly(new FluxContextWrite<>(this, ContextPropagation.contextCapture()));
+	}
+
+	/**
+	 * If Micrometer's {@code context-propagation} library is on the classpath, this
+	 * is a convenience shortcut to capture thread local values during the subscription phase
+	 * and put them in the {@link Context} that is visible upstream of this operator.
+	 * <p>
+	 * As a result this operator should generally be used as close as possible to the end of
+	 * the chain / subscription point.
+	 * <p>
+	 * The provided {@link Predicate} is used on keys associated to the thread locals
+	 * by the Context-Propagation API to filter which entries should be captured in the
+	 * first place.
+	 *
+	 * @param captureFilter a {@link Predicate} used on keys to determine if each entry should be
+	 * injected into the new {@link Context}
+	 * @return a new {@link Flux} where context-propagation API has been used to capture
+	 * entries which pass the filter and inject them into the {@link Context}
+	 * @see #contextCapture()
+	 */
+	public final Flux<T> contextCapture(Predicate<Object> captureFilter) {
```

Comment:
I wonder if we need the Predicate based variant to start, or until requested. I tend to think there will be more need for control when restoring while capture at the start of a chain would be more open about getting all that's available.